IntelLLVM does not support `-ipo1`

## Environment Information
Open Babel version: 7acf50c9d73f8016e15eb2a45ce74c8909152883
Operating system and version: EndeavourOS

## Expected Behavior
<!--  Describe the intended output or what you expected to see. -->

No build failure.

## Actual Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

Intel classic compilers, icc and icpc, are deprecated and removed, and now LLVM compilers, icx and icpx, are provided. Their compiler ID is IntelLLVM and thus matches Intel.

https://github.com/openbabel/openbabel/blob/7acf50c9d73f8016e15eb2a45ce74c8909152883/CMakeLists.txt#L325
https://github.com/openbabel/openbabel/blob/7acf50c9d73f8016e15eb2a45ce74c8909152883/CMakeLists.txt#L350-L351

However, `-ipo` of IntelLLVM compilers accepts no parameter, resulting in compilation error: `icpx: error: unknown argument: '-ipo1'`.

Also, `-Wl,-s` should be set for `CMAKE_EXE_LINKER_FLAGS_RELEASE` instead.

## Steps to Reproduce
<!--
 If the problem occurs with a particular file, please either upload and attach the file or include a link here - this greatly improves our ability to test your problem.
 Please include screenshots or text output if they help illustrate a behavior.
-->

```shellsession
$ CC=icx CXX=icpx cmake -D CMAKE_BUILD_TYPE=Release -S . -B build
$ cmake --build build
[  1%] Built target data_headers
[  1%] Building CXX object src/CMakeFiles/openbabel.dir/alias.cpp.o
icpx: error: unknown argument: '-ipo1'
icpx: warning: -Wl,-s: 'linker' input unused [-Wunused-command-line-argument]
make[2]: *** [src/CMakeFiles/openbabel.dir/build.make:76: src/CMakeFiles/openbabel.dir/alias.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1361: src/CMakeFiles/openbabel.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -347,8 +347,13 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES Intel AND UNIX)
     endif()
   endif()
 
-  set(CMAKE_C_FLAGS_RELEASE "-O3 -ipo1 -DNDEBUG -Wl,-s")
-  set(CMAKE_CXX_FLAGS_RELEASE "-O3 -ipo1 -DNDEBUG -Wl,-s")
+  if(POLICY CMP0069)
+    cmake_policy(SET CMP0069 NEW)
+    if("${CMAKE_BUILD_TYPE}" STREQUAL Release)
+      set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+    endif()
+  endif()
+  set(CMAKE_EXE_LINKER_FLAGS_RELEASE "-Wl,-s")
 
   # define WITH_SSE2 to enable the SSE2 instruction set.
   # Available on Pentium 4, Athlon 64, and newer CPUs.

